python/perf-kernels/fused_moe/moe-gemm.py

```diff
@@ -0,0 +1,495 @@
+import triton
+import torch
+import triton.language as tl
+import pytest
+from typing import Any, Dict, Optional
+import os
+import json
+import functools
+import argparse
+import sys
+
+SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
+PARENT_DIR = os.path.dirname(SCRIPT_DIR)  # This goes one level up from fused-moe/
+if PARENT_DIR not in sys.path:
+    sys.path.append(PARENT_DIR)
+
+from utils.benchmark_utils import get_available_models, get_model_configs  # noqa: E402
+
+M_THRESHOLD_SMALL = 256
+M_THRESHOLD_MEDIUM = 1024
```

Comment:
In general, for the moe layer, is there an assumption about the N dim? Like a range?
Why does the threshold a function of M only?

Reply:
If N here refers to the hidden size then for mistral 7B its 4096 and for 22B its 6144. 
https://github.com/ROCm/vllm/blob/develop/benchmarks/kernels/benchmark_moe.py
Here is the reference of the vllm impl of moe. Basically the idea is that we are changing the block size for the batch size (M) which is the biggest factor of changing work load. We got these number by calculating the memory vs compute bound on the roofline Mistral model.